FrontEnd/src/main/java/resourceDisplay/AssetsRepository.java

```diff
@@ -0,0 +1,329 @@
+
+        List<String> columnDisplays = new ArrayList<>();
+        ;
+        List<String> columnNames = new ArrayList<>();
+        List<String> columnTypes = new ArrayList<>();
+        ResultSet resultSetForAllAssets = runConfigQuery("forAllAssets", dbClient);
+
+        while (resultSetForAllAssets.next()) {
+            columnDisplays.add(resultSetForAllAssets.getString("columnDisplayName"));
+            columnNames.add(resultSetForAllAssets.getString("columnName"));
+            columnTypes.add(resultSetForAllAssets.getString("columnType"));
+        }
+
+        ResultSet resultSetForMainAssets = runConfigQuery("Main_Assets", dbClient);
+        while (resultSetForMainAssets.next()) {
+            columnDisplays.add(resultSetForMainAssets.getString("columnDisplayName"));
+            columnNames.add(resultSetForMainAssets.getString("columnName"));
+            columnTypes.add(resultSetForMainAssets.getString("columnType"));
+        }
+        String columns = String.join(", ", columnNames);
+        String statementString;
+        if (filterType != null) {
+            statementString = String.format("SELECT %s FROM Main_Assets WHERE %s = @%s"
+                    , columns, filterType, filterType);
+        } else {
+            statementString = String.format("SELECT %s FROM Main_Assets", columns);
+        }
+
+        TableQueryObject tableQueryObject = new TableQueryObject(columnDisplays, columnNames,
+                columnTypes, statementString);
+        return tableQueryObject;
+    }
+
+    /**
+     * This builds and executes the query for all assets
+     *
+     * @param dbClient - A client for connection to the DB
+     * @return A ResultListObject witch holds a list of display names for each column and a list
+     * of lists where each row holds information of one asset in strings (to use
+     * in template)
+     */
+    public ResultListObject getAllAssets(DatabaseClient dbClient) {
+        TableQueryObject tableQueryObject = createTableQueryObject(dbClient, null);
+        Statement statement = Statement.newBuilder(tableQueryObject.Query).build();
+        List<List<String>> results = executeQueryAndReturnList(statement, dbClient,
+                tableQueryObject.columnNames, tableQueryObject.columnTypes);
+        ResultListObject resultList = new ResultListObject(tableQueryObject.columnDisplays,
+                results);
+        return resultList;
+    }
+
+    /**
+     * This builds and executes the query with status filter
+     *
+     * @param dbClient - A client for connection to the DB
+     * @param status   - The wanted status from the user
+     * @return A ResultListObject witch holds a list of display names for each column and a list
+     * of lists where each row holds information of one asset in strings (to use
+     * in template)
+     */
+    public ResultListObject getAssetsByStatus(DatabaseClient dbClient, String status) {
```

Comment:
all three method here (by status/kind/location) share the exact same logic. if you ever want to change that logic, you'll have to do it in three different locations. To avoid that, we can do as follows: define a generic method called `getAssetsByCriteria(DatabaseClient dbClient, String status, String criteria)`. The `criteria` are will be `"location"\"status"\kind`, and then you can call the generic method with a one liner from the specific methods.